src/common/transformations/tests/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -533,3 +533,57 @@ TEST_F(TransformationTestsF, IfReverseInfer) {
         model_ref = std::make_shared<Model>(ResultVector{result}, ParameterVector{X, cond});
     }
 }
+
+TEST_F(TransformationTestsF, TransposeWithDynamicOrderReverseInfer) {
+    {
+        auto data = std::make_shared<opset10::Parameter>(element::dynamic, PartialShape::dynamic());
+        auto order = std::make_shared<opset10::Parameter>(element::i32, PartialShape::dynamic());
+        auto transpose = std::make_shared<opset10::Transpose>(data, order);
+        // Convolution is needed to produce static rank
+        auto weights =
+            opset10::Constant::create(element::f32, Shape{64, 3, 7, 7}, std::vector<float>(64 * 3 * 7 * 7, 0.1f));
+        auto conv = std::make_shared<opset10::Convolution>(transpose,
+                                                           weights,
+                                                           Strides{2, 2},
+                                                           CoordinateDiff{3, 3},
+                                                           CoordinateDiff{3, 3},
+                                                           Strides{1, 1});
+        auto result = std::make_shared<opset10::Result>(conv);
+        model = std::make_shared<Model>(ResultVector{result}, ParameterVector{data, order});
+        manager.register_pass<pass::ReverseShapeAndTypeInfer>();
+    }
+    {
+        auto data = std::make_shared<opset10::Parameter>(element::f32, PartialShape{DYN, 3, DYN, DYN});
```

Comment:
Why you have static shape[1] dimension? You don't know order of `Transpose` this should be pure dynamic rank.

Reply:
Fixed.